### PR TITLE
List applying EventPolicies in KafkaSink

### LINKED ISSUE #4039: [control-plane] List applying policies in KafkaSinks `.status.policies`
As the [Eventing Authorization](https://docs.google.com/document/d/1u6vyMviGpD13uyLKSGV7_D_p0ZRBPWg6PlLItRH8UFI/edit?resourcekey=0-gCE8brkIopDqc4jE7c9_tQ#heading=h.9zw0wrg4io53) feature track describes, target resources of `EventPolicies`, will reflect in their status, which `EventPolicy` is applied to them. 

For example:

```yaml
apiVersion: kafkasinks.eventing.knative.dev
kind: KafkaSink
metadata:
  name: my-sink
spec:
  ...
status:
  ...
  policies:
    - name: event-policy
      apiVersion: v1alpha1
    - name: another-event-policy
      apiVersion: v1alpha1

  conditions:
    - type: Ready
      status: "True"
    - type: EventPoliciesReady
      status: "True"
```

As the KafkaSink should support `EventPolicies`, wee need to do the following:

* Update the KafkaSink reconciler to show applying policies in the KafkaSinks `.status.policies` and set the KafkaSinks `EventPoliciesReady` condition. Also watch EventPolicies for changes.
* Set `EventPoliciesReady` condition to true, in case no policy applies, and set `message` according to the `default-authorization-mode` (e.g. `Default authorization mode is 'allow-all'`). Check the [Default Authorization Mode section](https://docs.google.com/document/d/1u6vyMviGpD13uyLKSGV7_D_p0ZRBPWg6PlLItRH8UFI/edit?resourcekey=0-gCE8brkIopDqc4jE7c9_tQ#heading=h.c6984bgldrz) in the feature track for additional information.

**Hints:**
* Requires #4069 

**Additional hints for new contributors before starting with this issue:**

1. When the issue has the `Draft` status, the issue is subject to change and thus should not be started to be worked on
2. Make sure you've read and understood the [CONTRIBUTING.md](https://github.com/knative-extensions/eventing-kafka-broker/blob/main/CONTRIBUTING.md) guidelines
3. Make sure you're able to [run Knative Eventing locally](https://github.com/knative/eventing/blob/main/DEVELOPMENT.md#create-a-cluster-and-a-repo) with [Knative Eventing Kafka](https://github.com/knative-extensions/eventing-kafka-broker/blob/main/DEVELOPMENT.md) and [run at least the unit tests](https://github.com/knative-extensions/eventing-kafka-broker/blob/main/DEVELOPMENT.md#run-unit-tests). 
4. Feel free to raise any questions you have either directly here in the issue, in the [#knative-eventing Slack channel](https://cloud-native.slack.com/archives/C04LMU33V1S) or join the [Eventing Workgroup Meeting / Office Hours](https://github.com/knative/community/blob/main/working-groups/WORKING-GROUPS.md#eventing)
5. When you feel comfortable with this issue, feel free to assign it to you (e.g. by commenting `/assign`). Please be aware that we might unassign you, if we don't see any progress from your side to give other contributors also a chance to work on this issue.

### PR DESCRIPTION
**Pull request by creydr**
Fixes #4039 

## Proposed Changes

- :gift: List applying EventPolicies in KafkaSink

**Release Note**
```release-note
List applying EventPolicies in KafkaSink
```

### CODE REVIEW

**Comment by creydr** [comment]
@pierDipi this is ready for review

**Comment by creydr** [comment]
@pierDipi can you recheck?

File: control-plane/pkg/core/config/utils.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func EventPoliciesFromAppliedEventPoliciesStatus(status duck.AppliedEventPolicie
 		eventPolicies = append(eventPolicies, contractPolicy)
 	}
 
-	if len(eventPolicies) == 0 {
+	if len(eventPolicies) == 0 && features.IsOIDCAuthentication() {
```
Comment (pierDipi) [inline control-plane/pkg/core/config/utils.go:100]: it seems that this condition needs to be true from the very beginning and return early ?

Comment (creydr) [reply control-plane/pkg/core/config/utils.go]: Updated it (anyhow, as we added only Ready EventPolicies and an EP only is Ready when OIDC is enabled this should result in the same - but this safes some cycles)

**Comment by creydr** [comment]
@pierDipi could you recheck?

**Review by pierDipi — APPROVED** [review]
/lgtm

**Pull request merged**